src/NuGet.Core/NuGet.Protocol/DependencyInfo/Utils.cs

```diff
@@ -30,52 +32,74 @@ public static VersionRange CreateVersionRange(string stringToParse)
             ILogger log,
             CancellationToken token)
         {
-            var index = await httpSource.GetJObjectAsync(
-                new HttpSourceRequest(registrationUri, log)
-                {
-                    IgnoreNotFounds = true
-                },
-                log,
-                token);
-
-            if (index == null)
+            using (var sourecCacheContext = new SourceCacheContext())
             {
-                // The server returned a 404, the package does not exist
-                return Enumerable.Empty<JObject>();
-            }
+                var httpSourceCacheContext = HttpSourceCacheContext.Create(sourecCacheContext, 0);
 
-            IList<Task<JObject>> rangeTasks = new List<Task<JObject>>();
+                var parts = registrationUri.OriginalString.Split(new char[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
+                var packageId = parts[parts.Length - 2];
```

Comment:
Consider moving this URI parsing code to a helper method, and also handling validation. It *should* always have the right format. You could also pass the id down instead.